tools/fs/files.js

```diff
@@ -737,6 +745,32 @@ files.extractTarGz = function (buffer, destPath, options) {
   }
 };
 
+function waitFSToSync(dir) {
+	let waitToSync = true;
+
+	while (waitToSync) {
+		try {
+			// After a successful unlink operation, sometimes windows fs takes longer than expected to sync
+			// so readdir would still return already deleted files
+			const filesInTempDirArray = files.readdir(dir);
+
+			// If stat doesn't throw a known error, it means that any queued operations have already been performed
+			// and committed, so the list returned by readdir is the actual file list of the directory
```

Comment:
Is there another (entirely different) way to detect this that we could consider?  Possibly a short-lived watch on the directory?  Something else?